cpp/src/arrow/flight/sql/odbc/odbc_impl/types.h

```diff
@@ -172,7 +172,7 @@ enum RowStatus : uint16_t {
 };
 
 struct MetadataSettings {
-  boost::optional<int32_t> string_column_length{boost::none};
+  std::optional<int32_t> string_column_length{std::nullopt};
```

Comment:
The default initialization is nullopt so there's no need to write it out explicitly